### ISSUE TITLE
Highlighting with VSCode

### ISSUE THREAD

**Issue by nqthqn** [philipbelesky/gray-matter#6]
First of all: I LOVE THIS THEME. ❤️ ❤️ ❤️  Thank you for making it.

So, I'm not sure if this is an issue with the theme or vscode, but I've got some funky things happening when I select parts of words:

I highlighted "color" but now it shows a blue box around "background-color"
<img width="322" alt="Screen Shot 2019-05-18 at 10 42 29 AM" src="https://user-images.githubusercontent.com/857356/57973183-b9eedc80-7959-11e9-9fc7-c51d18b54d37.png">

here I just clicked on the classname and it put this darker blue behind it making the gray text illegible:
<img width="191" alt="Screen Shot 2019-05-18 at 10 42 13 AM" src="https://user-images.githubusercontent.com/857356/57973184-b9eedc80-7959-11e9-9051-3c6eb57211d8.png">

**Comment by philipbelesky** [comment]
Great to hear you like the theme, and thanks for reporting the issue.

I only use the theme for writing markdown/prose (and assumed that's what others did!) so I haven't tested it for code syntax highlighting. That said I think this might be related to a disjunct between how Sublime and VSCode handles different selection highlights. I'll have a look in to this, but might not be able to get to it for a week. Also happy to accept a PR if you're interested into digging into things.

**Comment by nqthqn** [comment]
I'd love to open a PR - could you give me a few pointers? I presume I would clone this some place, have vscode use it and then change some color that relates to selection?

> Syntax highlighting is juvenile. When I was a child, I was taught 
arithmetic using colored rods 
(http://en.wikipedia.org/wiki/Cuisenaire_rods). I grew up and today I 
use monochromatic numerals. 

— Rob Pike

https://groups.google.com/forum/#!msg/golang-nuts/hJHCAaiL0so/kG3BHV6QFfIJ

**Comment by philipbelesky** [comment]
Yes, I think the easiest way to go is clone this directory and then test changes by copying over the json files to where the package is installed in VSCode. There probably is a cleaner way to install/test work with under-development plugins to VSCode but I've forgotten it.

The main thing to know when editing the theme is that the `build.sh` is responsible for automatically making the dark theme (via color substitution). So, essentially `Gray Matter Light.tmTheme` is the master copy.

I've been able to replicate the issue you mention. To address it, I think you will want to [look at the different scopes that VSCode uses](https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#scope-inspector) in that file identify the clashing scopes (class name vs selection-highlight).

As for how to resolve the clash, I'm open to a couple of options. Generally my feeling is that this theme should use the blue color very sparingly and for direct user action — i.e. the cursor, selected words, matching search results, etc. So in this case maybe the automatically code highlights for properties or classes could be demoted to one of the lesser gray colors.
